CMakeLists.txt

```diff
@@ -8,6 +8,8 @@
 # Check IF cmake has the required version
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11 FATAL_ERROR)
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
```

Comment:
It's not. This is for better tool support (automatic C++ completion in editors) + static code analysis. It should be merged as a separate PR.